lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -965,6 +965,12 @@ FIRRTLType firrtl::getDShlResult(FIRRTLType lhs, FIRRTLType rhs) {
   return IntType::get(lhs.getContext(), lhsi.isSigned(), width);
 }
 
+FIRRTLType firrtl::getDShlwResult(FIRRTLType lhs, FIRRTLType rhs) {
+  if (!lhs.isa<IntType>() || !rhs.isa<UIntType>())
```

Comment:
Are there any constraints between these two types when their widths are known, or is it really just two arbitrary width types of that sign?

Reply:
It looks there is no constraint as far as I see the firrtl implementation. @seldridge, is it correct?

Reply:
I'm not aware of any constraints here.

Reply:
ok cool!

Reply:
Thanks!